### PR TITLE
Merges extensions into one page

### PR DESCRIPTION
**Pull request by arcbtc**
Gets rid of "Manage Extensions" page and merges into "Extensions"

Check admin here https://extdemo.satoshigo.app/extensions?usr=2f1c28dd6ce44f7884edde1c8269b3c1

## Extra

- [x] merge `Manage Extensions` into `Extensions`
- [x] check if the extension version is compatible with the LNbits version
   - do not allow install if the versions are incompatible (check on both back-end and front-end)
   - new field for minimum required LNbits version: `"min_lnbits_version"`. And for release warning: `warning`

in`extensions.json`
```
        {
            "id": "watchonly",
            "repo": "https://github.com/lnbits/watchonly",
            "name": "Onchain Wallet",
            "version": "0.2.1",
            "min_lnbits_version": "0.10.7",
            "warning": "It is RECOMMENDED to set this extension as admin-only. Anyone with acceess can create any number of relays!",
            ...
        },
```
or in `config.json`:
```
{
  "name": "Onchain Wallet",
  "short_description": "Onchain watch only wallets",
  "tile": "/watchonly/static/bitcoin-wallet.png",
  "min_lnbits_version": "10.8",
  "warning": "This is NOT the official repo",
  "contributors": ["arcbtc", "motorina0"]
}
```
   - [x] show error message on the UI
     - image
   - [x] show warning message on the UI
      -  image

- [x] show `New Version` when a new version is available
  -  image

- [x] `i18n` text in `extensions.html`

### Tests
 - only admins can install, uninstall, upgrade, activate and deactivate extensions (regular users are not allowed)
 - installed extensions show their version on the card
 - setting the `min_lnbits_version` to a future LNbits version will not allow the installation
    - `extension.json` example [here] (https://github.com/motorina0/lnbits-extensions/blob/main/extensions.json#L355)
    - `config.json` (GitHub release) example [here](https://github.com/motorina0/watchonly/blob/main/config.json)
    - note: use the `Raw` path
 - setting the `min_lnbits_version` to a lower version (or empty) must allow the install
 - setting the `warning` field in the [extensions.json](https://github.com/motorina0/lnbits-extensions/blob/main/extensions.json#L377) or in the [config.json](https://github.com/motorina0/watchonly/blob/main/config.json#L6) will show the when the release is expanded
 - if a new version of an extension is added then the `New Version` badge is displayed

### CODE REVIEW

**Comment by motorina0** [comment]
Regular users can see the `Activate/Deactivate` toggle (only admins should):
image

**Comment by motorina0** [comment]
For some reason `Manage Extensions` is still visible on https://extdemo.satoshigo.app/. Works fine locally tough.

**Comment by motorina0** [comment]
UI nitpick: the cards have different heights depending on what buttons are visible:
image

**Comment by motorina0** [comment]
UI suggestion: show the version in a `q-badge`:

image

**Comment by motorina0** [comment]
Spinner still present after extension installed with success:
image

**Comment by arbadacarbaYK** [comment]
Shouldnt "Delete" only be shown once - or am I not getting a feature here ?

Bildschirm­foto 2023-04-29 um 14 42 51

**Comment by arcbtc** [comment]
> UI suggestion: show the version in a `q-badge`:
> 
> image

Space would be a real issue here. Having the number is all that is needed to reference when updating

**Comment by arcbtc** [comment]
> Shouldnt "Delete" only be shown once - or am I not getting a feature here ?
> 
> Bildschirm­foto 2023-04-29 um 14 42 51



> UI nitpick: the cards have different heights depending on what buttons are visible: image

This is a content issue. I dont understand why TPoS is missing a description. Please check again

**Comment by arcbtc** [comment]
Just checked and this is a different bug, for some reason descriptions are missing for some extensions

**Comment by motorina0** [comment]
> Manage extensions button leads to nowhere.

Cache issue. Button is no longer there.

**Comment by motorina0** [comment]
> Shouldnt "Delete" only be shown once - or am I not getting a feature here ?

- an extension can be listed on one or more repositories (see screenshot)
- the user can chose from which repo to install
- there is an `Uninstall` button on the dialog to quickly un-install the extesion
- if the user navigates to the repo&release from which the extension is already installed then it will show an `Uninstall` button instead of `Install`

image

File: lnbits/core/templates/core/extensions.html [hunk 1]
```diff
@@ -197,8 +204,11 @@
                 url_for('install.extensions')
               }}', '?usr=', g.user.id, '&enable=', extension.id].join('')"
             >
-              Enable
-              <q-tooltip>Enable extension for current user</q-tooltip></q-btn
+              {%raw%}{{ $t('enable') }}{%endraw%}
+              <q-tooltip
+                >{%raw%}{{ $t('enable_extension_details')
+                }}{%endraw%}</q-tooltip
+              ></q-btn
```
Comment (dni) [inline lnbits/core/templates/core/extensions.html:211]:
```suggestion
              <q-tooltip :v-text="$t('enable_extension_details')"></q-tooltip
              ></q-btn
```

Comment (dni) [reply lnbits/core/templates/core/extensions.html:211]: i think this is cleaner, without needed the raw

**Comment by arbadacarbaYK** [comment]
> > Shouldnt "Delete" only be shown once - or am I not getting a feature here ?
> 
>     * an extension can be listed on one or more repositories (see screenshot)
> 
>     * the user can chose from which repo to install
> 
>     * there is an `Uninstall` button on the dialog to quickly un-install the extesion
> 
>     * if the user navigates to the repo&release from which the extension is already installed then it will show an `Uninstall` button instead of `Install`
> 
> 
> image

But if one unistalls the version in one of the repos its still corresponding to deleting the whole extension, isnt it ? What would the state be where the extension is installed but no version from any repo attached ? 🤯 to me is confusing

**Comment by arbadacarbaYK** [comment]
IMO the features enable+activate+open and disable+uninstall are not clear enough. 

This would be my solution for the things brought up here

Tab Installed 

Bildschirm­foto 2023-05-03 um 20 25 34

- Only one link called "Extensions"
- Only two tabs: Installed, Manage (=All) whereby Manage only visable for admins 
- visualize the deactivated-status of the extensions with a darker background

***

Tab Manage

Bildschirm­foto 2023-05-03 um 20 48 08


- Visualize the version between name/desc
- Hightlight featured ones with a fixed 5 rating-stars in a different colour than not featured ratings
- order by features/name, installed/name, not active/name
- Show the version between the name and the desc 
- Call the former Button Manage->Modify and Open->Preview

*** 

PopUp Modify

Bildschirm­foto 2023-05-03 um 20 48 44

- Not only "Uninstall version" but also unclicking an active version should lead to "Do you want to permanently delete this version from your LNbits?" if no just disable version 
- Pressing "Uninstall Extension" button should lead to "Do you want to permanently deactivate this extension from LNbits?"
If no just disable extension

**Comment by motorina0** [comment]
- [x] merge `Manage Extensions` into `Extensions`
- [x] check if the extension version is compatible with the LNbits version
   - do not allow install if the versions are incompatible
   - check on both back-end and front-end
   - new field for minimum required LNbits version: `"min_lnbits_version"`
```
        {
            "id": "watchonly",
            "repo": "https://github.com/lnbits/watchonly",
            "name": "Onchain Wallet",
            "version": "0.2.1",
            "min_lnbits_version": "0.10.7",
            ...
        },
```
   - show error message on the UI
     - image
- [ ] check version compatibility for GitHub releases
- [x] add `warning` field for a particular release
```
        {
            "id": "nostrrelay",
            "repo": "https://github.com/lnbits/nostr-relay-extension",
            "name": "Nostr Relay",
            "warning": "It is RECOMMENDED to set this extension as admin-only. Anyone with acceess can create any number of relays!",
           ...
        },
```
   - [x] show warning message on the UI
      -  image
- [x] show `New Version` when a new version is available
  -  image

- [x] `i18n` text in `extensions.html`

**Comment by motorina0** [comment]
> IMO the features enable+activate+open and disable+uninstall are not clear enough.

@arbadacarbaYK thanks for the input. This is a pretty big change (UI wise). It can be discussed in a different PR/issue. (not included in this PR)

**Pull request merged**